docs/pulplift.md

```diff
@@ -91,6 +91,31 @@ If using libvirt, the [vagrant-sshfs](https://github.com/dustymabe/vagrant-sshfs
 vagrant up pulp3-source-centos7
 ```
 
+#### FIPS box pair
+
+The following boxes are meant to be run together as a pair:
+- pulp2-nightly-pulp3-source-fips-a (Pulp 3 FIPS **VM**)
+- pulp2-nightly-pulp3-source-fips-b (Pulp 2 FIPS **container** that runs **on top** of the "a" VM)
+
+
+To create or start them, this is the shortest command:
+```
+vagrant up pulp2-nightly-pulp3-source-fips-a && vagrant up --provider docker pulp2-nightly-pulp3-source-fips-b || vagrant up --provider docker pulp2-nightly-pulp3-source-fips-b
```

Comment:
@mikedep333 after running these commands, I got:
```

Bringing machine 'pulp2-nightly-pulp3-source-fips-b' up with 'docker' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

docker provider:
* One of "build_dir", "git_repo" or "image" must be set


Bringing machine 'pulp2-nightly-pulp3-source-fips-b' up with 'docker' provider...
There are errors in the configuration of this machine. Please fix
the following errors and try again:

docker provider:
* One of "build_dir", "git_repo" or "image" must be set
```

Reply:
That makes no sense. We have `image: quay.io/mikedep333/vagrant-pulp2:latest` in `vagrant/boxes.d/30-source.yaml`

Reply:
oh! I need to update forklift

Reply:
maybe we need to update the doc with something like: 
```
make sure your pulplift/forklift is updated
```

Reply:
tests are failing:
```
E           pulp_smash.exceptions.CalledProcessError: Command ('sudo', 'bash', '/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh', 'localhost', 'rpm_base_4repos') returned non-zero exit status 127.
E           
E           stdout: MongoDB shell version v3.4.24
E           connecting to: mongodb://localhost:27017/pulp_database
E           MongoDB server version: 3.4.24
E           { "ok" : 1 }
E           
E           
E           stderr: /home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh: line 13: mongorestore: command not found
```

Reply:
> `make sure your pulplift/forklift is updated`

We'll put that in the announcement email.

Reply:
> tests are failing:
> 
> ```
> E           pulp_smash.exceptions.CalledProcessError: Command ('sudo', 'bash', '/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh', 'localhost', 'rpm_base_4repos') returned non-zero exit status 127.
> E           
> E           stdout: MongoDB shell version v3.4.24
> E           connecting to: mongodb://localhost:27017/pulp_database
> E           MongoDB server version: 3.4.24
> E           { "ok" : 1 }
> E           
> E           
> E           stderr: /home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh: line 13: mongorestore: command not found
> ```
> 
> https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh

Pushing a fix momentarily to host-pulp2.yml.

You can quickly update your a box by running:
`sudo dnf install mongodb-org-tools`

Reply:
I got 2 errors now, one is FIPS related:
```
pulp_2to3_migration/tests/functional/file_base.py:72: in run_migration
    task = monitor_task(mp_run_response.task)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

task_href = '/pulp/api/v3/tasks/9b3c5e4d-aa50-4ffb-a3ac-3239345fa5de/'

    def monitor_task(task_href):
        """Polls the Task API until the task is in a completed state.
    
        Prints the task details and a success or failure message. Exits on failure.
    
        Args:
            task_href(str): The href of the task to monitor
    
        Returns:
            list[str]: List of hrefs that identify resource created by the task
    
        """
        completed = ["completed", "failed", "canceled"]
        task = tasks.read(task_href)
        while task.state not in completed:
            sleep(SLEEP_TIME)
            task = tasks.read(task_href)
    
        if task.state != "completed":
>           raise PulpTaskError(task=task)
E           pulp_smash.pulp3.bindings.PulpTaskError: (PulpTaskError(...), 'Pulp task failed ([digital envelope routines: EVP_DigestInit_ex] disabled for FIPS)')
```
the other is mongo related:
```
E           pulp_smash.exceptions.CalledProcessError: Command ('sudo', 'bash', '/home/vagrant/devel/pulp-2to3-migration/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh', 'localhost', 'rpm_base_4repos') returned non-zero exit status 1.
E           
E           stdout: MongoDB shell version v3.4.24
E           connecting to: mongodb://localhost:27017/pulp_database
E           MongoDB server version: 3.4.24
E           { "dropped" : "pulp_database", "ok" : 1 }
E           
E           
E           stderr: 2021-03-18T20:47:10.284+0000	preparing collections to restore from
E           2021-03-18T20:47:10.292+0000	reading metadata for pulp_database.repo_content_units from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.323+0000	restoring pulp_database.repo_content_units from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.331+0000	reading metadata for pulp_database.lazy_content_catalog from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.364+0000	restoring pulp_database.lazy_content_catalog from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.367+0000	reading metadata for pulp_database.content_types from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.370+0000	restoring indexes for collection pulp_database.repo_content_units from metadata
E           2021-03-18T20:47:10.418+0000	finished restoring pulp_database.repo_content_units (73 documents)
E           2021-03-18T20:47:10.457+0000	restoring pulp_database.content_types from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.460+0000	reading metadata for pulp_database.units_rpm from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.530+0000	restoring pulp_database.units_rpm from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.534+0000	reading metadata for pulp_database.units_modulemd from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.538+0000	restoring indexes for collection pulp_database.content_types from metadata
E           2021-03-18T20:47:10.540+0000	restoring indexes for collection pulp_database.lazy_content_catalog from metadata
E           2021-03-18T20:47:10.556+0000	restoring indexes for collection pulp_database.units_rpm from metadata
E           2021-03-18T20:47:10.590+0000	restoring pulp_database.units_modulemd from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.659+0000	finished restoring pulp_database.lazy_content_catalog (58 documents)
E           2021-03-18T20:47:10.659+0000	reading metadata for pulp_database.units_erratum from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.718+0000	finished restoring pulp_database.content_types (24 documents)
E           2021-03-18T20:47:10.718+0000	finished restoring pulp_database.units_rpm (35 documents)
E           2021-03-18T20:47:10.754+0000	restoring pulp_database.units_erratum from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.760+0000	reading metadata for pulp_database.repo_distributors from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.787+0000	restoring pulp_database.repo_distributors from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.789+0000	reading metadata for pulp_database.erratum_pkglists from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.815+0000	restoring pulp_database.erratum_pkglists from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:10.817+0000	restoring indexes for collection pulp_database.units_erratum from metadata
E           2021-03-18T20:47:10.817+0000	restoring indexes for collection pulp_database.repo_distributors from metadata
E           2021-03-18T20:47:10.817+0000	restoring indexes for collection pulp_database.units_modulemd from metadata
E           2021-03-18T20:47:10.818+0000	restoring indexes for collection pulp_database.erratum_pkglists from metadata
E           2021-03-18T20:47:11.011+0000	finished restoring pulp_database.units_erratum (8 documents)
E           2021-03-18T20:47:11.011+0000	reading metadata for pulp_database.migration_trackers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.084+0000	finished restoring pulp_database.units_modulemd (10 documents)
E           2021-03-18T20:47:11.118+0000	finished restoring pulp_database.erratum_pkglists (8 documents)
E           2021-03-18T20:47:11.136+0000	finished restoring pulp_database.repo_distributors (10 documents)
E           2021-03-18T20:47:11.159+0000	restoring pulp_database.migration_trackers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.160+0000	reading metadata for pulp_database.workers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.185+0000	restoring pulp_database.workers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.186+0000	reading metadata for pulp_database.task_status from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.210+0000	restoring pulp_database.task_status from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.211+0000	reading metadata for pulp_database.repo_importers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.211+0000	no indexes to restore
E           2021-03-18T20:47:11.211+0000	finished restoring pulp_database.workers (6 documents)
E           2021-03-18T20:47:11.211+0000	restoring indexes for collection pulp_database.migration_trackers from metadata
E           2021-03-18T20:47:11.226+0000	finished restoring pulp_database.migration_trackers (7 documents)
E           2021-03-18T20:47:11.250+0000	restoring pulp_database.repo_importers from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.250+0000	restoring indexes for collection pulp_database.task_status from metadata
E           2021-03-18T20:47:11.251+0000	reading metadata for pulp_database.units_package_group from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.362+0000	finished restoring pulp_database.task_status (6 documents)
E           2021-03-18T20:47:11.396+0000	restoring pulp_database.units_package_group from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.399+0000	reading metadata for pulp_database.repos from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.453+0000	restoring pulp_database.repos from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.455+0000	reading metadata for pulp_database.permissions from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.455+0000	restoring indexes for collection pulp_database.units_package_group from metadata
E           2021-03-18T20:47:11.456+0000	restoring indexes for collection pulp_database.repo_importers from metadata
E           2021-03-18T20:47:11.490+0000	restoring pulp_database.permissions from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.519+0000	restoring indexes for collection pulp_database.repos from metadata
E           2021-03-18T20:47:11.708+0000	finished restoring pulp_database.units_package_group (5 documents)
E           2021-03-18T20:47:11.708+0000	reading metadata for pulp_database.units_srpm from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.727+0000	finished restoring pulp_database.repo_importers (5 documents)
E           2021-03-18T20:47:11.756+0000	finished restoring pulp_database.repos (5 documents)
E           2021-03-18T20:47:11.794+0000	restoring pulp_database.units_srpm from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.800+0000	reading metadata for pulp_database.units_package_langpacks from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.836+0000	restoring pulp_database.units_package_langpacks from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.837+0000	restoring indexes for collection pulp_database.units_srpm from metadata
E           2021-03-18T20:47:11.843+0000	reading metadata for pulp_database.units_modulemd_defaults from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.894+0000	restoring indexes for collection pulp_database.units_package_langpacks from metadata
E           2021-03-18T20:47:11.894+0000	finished restoring pulp_database.units_srpm (3 documents)
E           2021-03-18T20:47:11.894+0000	restoring indexes for collection pulp_database.permissions from metadata
E           2021-03-18T20:47:11.932+0000	restoring pulp_database.units_modulemd_defaults from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.939+0000	reading metadata for pulp_database.repo_publish_results from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:11.961+0000	finished restoring pulp_database.units_package_langpacks (3 documents)
E           2021-03-18T20:47:12.016+0000	restoring pulp_database.repo_publish_results from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.018+0000	reading metadata for pulp_database.units_package_category from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.018+0000	finished restoring pulp_database.permissions (4 documents)
E           2021-03-18T20:47:12.018+0000	restoring indexes for collection pulp_database.units_modulemd_defaults from metadata
E           2021-03-18T20:47:12.068+0000	finished restoring pulp_database.units_modulemd_defaults (3 documents)
E           2021-03-18T20:47:12.093+0000	restoring pulp_database.units_package_category from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.103+0000	reading metadata for pulp_database.repo_sync_results from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.135+0000	restoring pulp_database.repo_sync_results from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.136+0000	restoring indexes for collection pulp_database.repo_publish_results from metadata
E           2021-03-18T20:47:12.140+0000	reading metadata for pulp_database.units_distribution from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.163+0000	restoring indexes for collection pulp_database.units_package_category from metadata
E           2021-03-18T20:47:12.164+0000	restoring indexes for collection pulp_database.repo_sync_results from metadata
E           2021-03-18T20:47:12.198+0000	restoring pulp_database.units_distribution from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.325+0000	finished restoring pulp_database.units_package_category (3 documents)
E           2021-03-18T20:47:12.325+0000	reading metadata for pulp_database.resource_manager_lock from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.345+0000	finished restoring pulp_database.repo_publish_results (3 documents)
E           2021-03-18T20:47:12.377+0000	restoring pulp_database.resource_manager_lock from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.379+0000	finished restoring pulp_database.repo_sync_results (3 documents)
E           2021-03-18T20:47:12.382+0000	reading metadata for pulp_database.units_package_environment from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.420+0000	restoring pulp_database.units_package_environment from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.420+0000	restoring indexes for collection pulp_database.units_distribution from metadata
E           2021-03-18T20:47:12.420+0000	restoring indexes for collection pulp_database.resource_manager_lock from metadata
E           2021-03-18T20:47:12.424+0000	reading metadata for pulp_database.users from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.450+0000	finished restoring pulp_database.resource_manager_lock (1 document)
E           2021-03-18T20:47:12.618+0000	finished restoring pulp_database.units_distribution (1 document)
E           2021-03-18T20:47:12.654+0000	restoring pulp_database.users from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.658+0000	reading metadata for pulp_database.roles from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.693+0000	restoring pulp_database.roles from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.700+0000	reading metadata for pulp_database.celery_beat_lock from archive 'pulp-2to3-migration-test-fixtures/rpm_base_4repos/mongodb.rpm_base_4repos.archive'
E           2021-03-18T20:47:12.701+0000	restoring indexes for collection pulp_database.users from metadata
E           2021-03-18T20:47:12.702+0000	Failed: pulp_database.celery_beat_lock: error creating collection pulp_database.celery_beat_lock: error running create command: a collection 'pulp_database.celery_beat_lock' already exists
```

Reply:
Per our IRC discussion:

The 1st bug looks like a FIPS error in the application

The 2nd bug looks like a bug in the test case, possibly the application.

A bug in the vagrant environment is expected to cause errors like "mongo cannot connect" or "command not found".

So we should merge.

Reply:
@bmbouter @daviddavis @gerrod3 @ggainey @pavelpicka ^
I believe the 2nd test is related to migration plugin script to set the test, probably the script needs some rewriting: https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/tests/functional/scripts/set_pulp2.sh